backend/src/nodes/properties/inputs/pytorch_inputs.py

```diff
@@ -20,9 +29,46 @@ def __init__(
     ):
         super().__init__(input_type, label)
 
-    def enforce(self, value):
+    def enforce(self, value: PyTorchModel):
+        if torch is not None:
+            assert isinstance(value, torch.nn.Module), "Expected a PyTorch model."
+            assert isPyTorchModel(value), "Expected a supported PyTorch model."
+        return value
+
+
+class SrModelInput(ModelInput):
+    def __init__(
+        self,
+        label: str = "Model",
+        input_type: Union[
+            str, ExpressionJson
+        ] = "PyTorchModel { arch: invStrSet(PyTorchModel::FaceArchs) }",
+    ):
+        super().__init__(label, input_type)
+
+    def enforce(self, value: PyTorchSRModel):
+        if torch is not None:
+            assert isinstance(value, torch.nn.Module), "Expected a PyTorch model."
+            assert isPyTorchSRModel(value), "Expected a regular Super-Resolution model."
+        return value
+
+
+class FaceModelInput(ModelInput):
+    def __init__(
+        self,
+        label: str = "Face SR Model",
+        input_type: Union[
+            str, ExpressionJson
+        ] = "PyTorchModel { arch: PyTorchModel::FaceArchs }",
```

Comment:
There is a better way for doing this. We always want the input type to be a subset of `PyTorchModel { arch: PyTorchModel::FaceArchs }`, but the dev can just specify any type right now. Instead, I suggest using an intersection.

```py
input_type: ExpressionJson = "PyTorchModel"
super().__init__(label, intersect(input_type, "PyTorchModel { arch: PyTorchModel::FaceArchs }"))
```